### PR TITLE
Completes 8. PATCH/api/articles/article_id

### PR DESCRIPTION
**Pull request by Tayamul**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -187,31 +187,157 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    const article_id = 1;
+    const newVote = 67;
+    const inc = { incVote: newVote };
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send(inc)
+      .expect(200)
+      .then(({ body: { votes } }) => {
+        expect(votes[0].votes).toBe(167);
+        votes.forEach((vote) => {
```
Comment (popatre) [inline __tests__/app.test.js]: It should respond with the updated article
So should have a key of `article` and be a single object - but with the new votes added

At the moment, it looks to be an array

File: __tests__/app.test.js [hunk 2]
```diff
@@ -187,31 +187,157 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    const article_id = 1;
+    const newVote = -67;
+    const inc = { incVote: newVote };
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send(inc)
+      .expect(200)
+      .then(({ body: { votes } }) => {
+        expect(votes[0].votes).toBe(33);
+        votes.forEach((vote) => {
```
Comment (popatre) [inline __tests__/app.test.js]: send back single object please

File: __tests__/app.test.js [hunk 3]
```diff
@@ -187,31 +187,157 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    const inc = { incVote: newVote };
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send(inc)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not Found In The Database");
+      });
+  });
+  test("400: newVote requested by the client is a string", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: description wise - should be inc_votes

File: __tests__/app.test.js [hunk 4]
```diff
@@ -187,31 +187,157 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+              votes: expect.any(Number),
+            })
+          );
+        });
+      });
+  });
+  test("200: decrements votes by the given amount for a certain article requested by the client", () => {
+    const article_id = 1;
+    const newVote = -67;
+    const inc = { incVote: newVote };
```
Comment (popatre) [inline __tests__/app.test.js]: The ticket wants it like `{ inc_votes : 1 }`
with the underscore

File: controllers/articles.controller.js [hunk 5]
```diff
@@ -34,4 +35,20 @@ const getArticlesCommentsById = (req, res, next) => {
...
 };
 
-module.exports = { getArticles, getArticlesById, getArticlesCommentsById };
+const patchArticlesById = (req, res, next) => {
+  const { article_id } = req.params;
+  const { incVote } = req.body;
+
+  updateArticles(article_id, incVote)
+    .then((votes) => {
+      res.status(200).send({ votes });
```
Comment (popatre) [inline controllers/articles.controller.js]: needs a key of `article` - it shoulld be the updated article - with the new votes

File: controllers/articles.controller.js [hunk 6]
```diff
@@ -34,4 +35,20 @@ const getArticlesCommentsById = (req, res, next) => {
     .catch((err) => next(err));
 };
 
-module.exports = { getArticles, getArticlesById, getArticlesCommentsById };
+const patchArticlesById = (req, res, next) => {
+  const { article_id } = req.params;
+  const { incVote } = req.body;
```
Comment (popatre) [inline controllers/articles.controller.js]: inc_votes

File: models/articles.model.js [hunk 7]
```diff
@@ -63,8 +63,32 @@ const selectCommentsByArticles = (article_id) => {
...
+  return db
+    .query(queryString, [incrementBy, article_id])
+    .then(({ rows, rowCount }) => {
+      if (rowCount === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "Not Found In The Database",
+        });
+      }
+      return rows;
```
Comment (popatre) [inline models/articles.model.js]: return the single object, not an array

File: models/articles.model.js [hunk 8]
```diff
@@ -63,8 +63,32 @@ const selectCommentsByArticles = (article_id) => {
...
+  }
+  const queryString = `
+  UPDATE articles
+  SET votes = votes + $1
+  WHERE article_id = $2
+  RETURNING *;`;
+
+  return db
+    .query(queryString, [incrementBy, article_id])
+    .then(({ rows, rowCount }) => {
```
Comment (popatre) [inline models/articles.model.js:118]: nice destructuring

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 9]
```diff
@@ -187,31 +187,148 @@ describe("GET/api/articles/:article_id/comments", () => {
...
+    const article_id = 2;
+
+    const inc = { noVoteRequested: 45 };
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send(inc)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
```
Comment (B7Bprog) [inline __tests__/app.test.js:454]: Could add another test (400) for missing body. (`inc = {};`) .

**Review by B7Bprog — APPROVED** [review]
Hey Tayamul, very nice and thorough testing, I added one more case that you could write a test for. Please consider adding it, then feel free to merge.

**Pull request merged**